File: Algorithms/Sorting-Algorithms/C++/insertion_sort.cpp
#include <iostream>

void insertion_sort(int array[], int length){
	for(int i = 0; i < length; i++){
		int key = array[i];
		int j = i-1;
		while (j > 0 && key < array[j]){
			array[j+1] = array[j];
			j--;
		}
		array[j+1] = key; 
	}
}

void print_array(int array[], int length)
{
    int index;
    for (index = 0; index < length; index++)
        std::cout << array[index] << " ";
    std::cout << std::endl;
}

int main(){
	int array[] = {10, 30, 20, 70, 40, 50, 60};
	int length = (sizeof(array)/sizeof(int));		
	insertion_sort(array, length);
	print_array(array, length);
}

File: Algorithms/Sorting_Algorithms/JS/InsertionSort.js
function InsertionSort(arr){
    for (let i = 1; i < arr.length; i++){
        var key = arr[i]
        var j = i - 1;
        for(;j >= 0 && key < arr[j];)
            arr[j+1] = arr[j--];
        arr[j+1] = key;
    }
}

function PrintArray(arr){
    sb = ""
    arr.forEach(element => {
        sb += ` ${element}`;
    });
     console.log(sb);
}

function main(){
    console.log("Insertion Sort");
    array = [30, 40, 10, 60, 50, 80, 70, 20];
    console.log("Original vector: ");
    PrintArray(array);
    InsertionSort(array);
    console.log("Sorted vector: ");
    PrintArray(array);
    console.log("finished...");
}

main();
